Makefile

```diff
@@ -1,43 +1,4 @@
-.PHONY: commit-check check clean lint style test test-watch coverage circleci circleci-memory prompt-clean prompt-long-process cachix-push-all
```

Comment:
I guess this will be the most controversial... Thoughts?

Reply:
Bold, I like it :-) We still want to keep stack as a first class citizen for development, but I don't think this hurts that goal. 

Suggest that we even completely kill the Makefile, `cachix-push-all` can be something that we note in the nix/README or add as e.g. 'postgrest-cachix-push' (to be then also used in CI). Its use will always be reserved to a very limited group anyway.

This will also be an opportunity to streamline the postgrest-docs on running tests.

Reply:
> Bold, I like it :-) We still want to keep stack as a first class citizen for development, but I don't think this hurts that goal.

stack for development (i.e. including tools) or just to be able to build from source?

I am not against keeping "stack for development", but mainly because I have no idea about that. So please help my understanding here: For me, developing for PostgREST - what kind of things are not possible for me to do with the nix devtools, that I need stack for?

> Suggest that we even completely kill the Makefile, `cachix-push-all` can be something that we note in the nix/README or add as e.g. 'postgrest-cachix-push' (to be then also used in CI). Its use will always be reserved to a very limited group anyway.

Perfect.

> This will also be an opportunity to streamline the postgrest-docs on running tests.

Even more so!

Reply:
The nix dev environment covers everything and is the preferred setup. 

However, we don't want to lock out one-off contributors if they don't use Nix, so we want to keep building and running basic tests with stack functional. We cover this with the stack.yaml and tests/with_tmp_db script, ensuring that it still works with the one CI job. Linting/styling the haskell source code correctly will be a bit more tricky for non-nix-users when we remove the Makefile, but we'll always have our CI to quickly flag issues if needed.

Reply:
One more reason to keep stack. robx receives notifications from them to keep dependencies up-to-date. See:

- https://github.com/PostgREST/postgrest/pull/1611#issue-498341270
- https://github.com/commercialhaskell/stackage/issues/5672.

Really helpful for us. stack upgrades GHC on their LTSs so this lets us know what we need to do to upgrade.

Reply:
I would not be opposed to killing :skull:  the Makefile though, I only use Nix for development now :)

Reply:
Just killed the Makefile! Felt good!